craft_parts/plugins/uv_plugin.py

```diff
@@ -0,0 +1,130 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2020-2024 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""The uv plugin."""
+
+import shlex
+import shutil
+from typing import Literal, cast
+
+import pydantic
+from overrides import override
+
+from craft_parts import errors
+from craft_parts.plugins import validator
+
+from .base import BasePythonPlugin
+from .properties import PluginProperties
+
+
+class UvPluginProperties(PluginProperties, frozen=True):
+    """The part properties used by the uv plugin."""
+
+    plugin: Literal["uv"] = "uv"
+
+    uv_extras: set[str] = pydantic.Field(
```

Comment:
We also have dependency groups and potentially future options. I'm still unsure about this vs a `uv_extra_args` option.

Reply:
I think I lean towards adding `uv_groups` rather than a generic `uv_extra_args` for both - the former is a lot easier to document (both for us and a user).

Reply:
I'm fine with that. Let's do it!